### PR TITLE
L shape fix

### PR DESCRIPTION
**Pull request by ConorLeckey**
-Added logic to fill in 'L' tiles 
-Added logic to fill in corner tiles

### CODE REVIEW

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 1]
```diff
@@ -102,11 +104,38 @@ TilemanModeConfig provideConfig(ConfigManager configManager) {
...
+            {MovementFlag.BLOCK_MOVEMENT_NORTH_WEST,
+                    MovementFlag.BLOCK_MOVEMENT_WEST,
+                    MovementFlag.BLOCK_MOVEMENT_SOUTH_WEST};
+
+    private final MovementFlag[] directionalEast = new MovementFlag[]
+            {MovementFlag.BLOCK_MOVEMENT_NORTH_EAST,
+                    MovementFlag.BLOCK_MOVEMENT_EAST,
+                    MovementFlag.BLOCK_MOVEMENT_SOUTH_EAST};
+
+    private final MovementFlag[] allDirections = Stream.of(directionalNorth, directionalSouth, directionalWest, directionalEast).flatMap(Stream::of).toArray(MovementFlag[]::new);
```
Comment (sram1337) [inline src/main/java/com/tileman/TilemanModePlugin.java:133]: This will duplicate every corner twice in the allDirections list.

Comment (ConorLeckey) [reply src/main/java/com/tileman/TilemanModePlugin.java:133]: Good point :)

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 2]
```diff
@@ -410,17 +429,147 @@ private void handleWalkedToTile(LocalPoint currentPlayerPoint) {
...
+            int yDiff = currentPlayerPoint.getY() - lastTile.getY();
             int yModifier = yDiff / 2;
             int xModifier = xDiff / 2;
 
-            updateTileMark(new LocalPoint(currentPlayerPoint.getX() + xModifier, currentPlayerPoint.getY() + yModifier), true);
+            switch(lastTile.distanceTo(currentPlayerPoint)) {
+                case 0:
+                    return;
+                case 256:
+                case 362:
```
Comment (sram1337) [inline src/main/java/com/tileman/TilemanModePlugin.java:442]: can we add comments describing which cases these numbers represent?

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 3]
```diff
@@ -410,17 +429,147 @@ private void handleWalkedToTile(LocalPoint currentPlayerPoint) {
...
+        } else {
+            tileBesideXDiff = 0;
+            tileBesideYDiff = yDiff;
+        }
+
+        MovementFlag[] tileBesideFlagsArray = getTileMovementFlags(lastTile.getX() + tileBesideXDiff, lastTile.getY() + tileBesideYDiff);
+
+        if (tileBesideFlagsArray.length == 0) {
+            updateTileMark(new LocalPoint(lastTile.getX() + tileBesideXDiff / 2, lastTile.getY() + tileBesideYDiff / 2), true);
+        } else if (containsAnyOf(tileBesideFlagsArray, fullBlock) || !containsAnyOf(allDirections, tileBesideFlagsArray)) {
```
Comment (sram1337) [inline src/main/java/com/tileman/TilemanModePlugin.java:473]: What is this `if` checking? Seems like it should be `containsAnyOf(tileBesideFlagsArray, fullBlock) || containsAnyOf(allDirections, tileBesideFlagsArray)`, ie "if it contains a blocking object"

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 4]
```diff
@@ -410,17 +429,147 @@ private void handleWalkedToTile(LocalPoint currentPlayerPoint) {
...
+            if (yModifier == 64) {
+                yModifier = 128;
+            } else if (xModifier == 64) {
+                xModifier = 128;
+            }
+            updateTileMark(new LocalPoint(lastTile.getX() + xModifier, lastTile.getY() + yModifier), true);
+        } else {
```
Comment (sram1337) [inline src/main/java/com/tileman/TilemanModePlugin.java:480]: can we add a comment describing which case this is handling?

**Comment by sram1337** [comment]
Works great! I'm fine with merging this, but I'd like to understand it better. I had trouble following what was happening in the code

File: src/main/java/com/tileman/TilemanModePlugin.java (hunk 3 above)
Comment (ConorLeckey) [reply src/main/java/com/tileman/TilemanModePlugin.java:473]: It's just checking if there is a tile that is fully blocked, tbh it is poorly laid out so I'll move the logic about

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 5]
```diff
@@ -410,17 +429,147 @@ private void handleWalkedToTile(LocalPoint currentPlayerPoint) {
...
+            tileBesideXDiff = 0;
+            tileBesideYDiff = yDiff;
+        }
+
+        MovementFlag[] tileBesideFlagsArray = getTileMovementFlags(lastTile.getX() + tileBesideXDiff, lastTile.getY() + tileBesideYDiff);
+
+        if (tileBesideFlagsArray.length == 0) {
+            updateTileMark(new LocalPoint(lastTile.getX() + tileBesideXDiff / 2, lastTile.getY() + tileBesideYDiff / 2), true);
+        } else if (containsAnyOf(tileBesideFlagsArray, fullBlock) || !containsAnyOf(allDirections, tileBesideFlagsArray)) {
+            if (yModifier == 64) {
```
Comment (sram1337) [inline src/main/java/com/tileman/TilemanModePlugin.java:474]: when would yModifier be 64? Isn't that like half a tile?

Comment (ConorLeckey) [reply src/main/java/com/tileman/TilemanModePlugin.java:474]: When it's one tile, the modifier is the difference / 2 and a diff of 128 is one tile.

File: src/main/java/com/tileman/TilemanModePlugin.java (hunk 4 above)
Comment (ConorLeckey) [reply src/main/java/com/tileman/TilemanModePlugin.java]: Changed the logic, should be more readable now without comment

**Pull request merged**